p2p/src/network/identify/stream_effectful/p2p_network_identify_stream_effectful_actions.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{ConnectionAddr, P2pAction, P2pState, PeerId, StreamId};
+use openmina_core::ActionEvent;
+use redux::EnablingCondition;
+use serde::{Deserialize, Serialize};
+
+/// Identify stream related actions.
+#[derive(Debug, Clone, Serialize, Deserialize, ActionEvent)]
+pub enum P2pNetworkIdentifyStreamEffectfulAction {
+    /// Creates a new stream state.
+    SendIdentify {
+        addr: ConnectionAddr,
+        peer_id: PeerId,
+        stream_id: StreamId,
+    },
+}
+
+macro_rules! enum_field {
```

Comment:
Do we need this macro just for 3 fields?

Reply:
It is not need, but I left it there just in case.